modules/common/affinity/affinity.go

```diff
@@ -54,3 +62,78 @@ func DistributePods(
 		},
 	}
 }
+
+// DistributePodsOverride - returns rule to ensure that two replicas of the same selector
+// should not run if possible on the same worker node
+func DistributePodsOverride(
+	selectorKey string,
+	selectorValues []string,
+	topologyKey string,
+	overrides *OverrideSpec,
+) (*corev1.Affinity, error) {
+	defaultAffinity := &corev1.Affinity{
+		PodAntiAffinity: &corev1.PodAntiAffinity{
+			// This rule ensures that two replicas of the same selector
+			// should not run if possible on the same worker node
+			PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+				{
+					PodAffinityTerm: corev1.PodAffinityTerm{
+						LabelSelector: &metav1.LabelSelector{
+							MatchExpressions: []metav1.LabelSelectorRequirement{
+								{
+									Key:      selectorKey,
+									Operator: metav1.LabelSelectorOpIn,
+									Values:   selectorValues,
+								},
+							},
+						},
+						// usually corev1.LabelHostname "kubernetes.io/hostname"
+						// https://github.com/kubernetes/api/blob/master/core/v1/well_known_labels.go#L20
+						TopologyKey: topologyKey,
+					},
+					Weight: 100,
+				},
+			},
+		},
+	}
+	// patch the default affinity Object with the data passed as input
+	if overrides != nil {
+		patchedAffinity, err := toCoreAffinity(defaultAffinity, overrides)
+		return patchedAffinity, err
+	}
+	return defaultAffinity, nil
+}
+
+func toCoreAffinity(
+	affinity *v1.Affinity,
+	override *OverrideSpec,
```

Comment:
I'm still not clear why this arg needs to be an OverrideSpec rather than just another v1.Affinity structure. Maybe my eyes simply aren't detecting the difference between the two structures?

Reply:
Replying to myself, this is discussed in another comment so no need to discuss it again, here.

Reply:
As mentioned in my previous comment, the redundancy introduced by the struct is mainly related to make sure we control future updates to what we support in the `Topology` Spec (it includes `affinity.OverrideSpec`). We can switch to `corev1.Affinity` (there is no difference between the two structs atm), but we need to be aware that we can get updates w/o our control. 
If we're good with that implication, I can `s/OverrideSpec/corev1.Affinity/g`.